core/model/job/step/gotolocation/fsm.go

```diff
@@ -0,0 +1,24 @@
+package gotolocation
+
+import (
+	"fmt"
+
+	"github.com/cocoonspace/fsm"
+)
+
+func checkTransition(current State, event Event) error {
+	f := fsm.New(fsm.State(current))
+	f.Transition(fsm.On(fsm.Event(MoveEvent)), fsm.Src(fsm.State(New)), fsm.Dst(fsm.State(Moving)))
```

Comment:
This isn't as readable as I'd like. I'd like to be able to read the state transitions next to each other with as little cruft as possible.

```
  [NearEvent, Moving, Near],
  [ArrivedEvent, Near, Arrived],
  ... 
```

Not a huge deal for now.

Reply:
i am not a fan of type conversion either, but we use generic fsm library with its own Sate/Events types. 
Alternative could be using fsm.State in our domain packages. 
I like this implementation better because it does not leak fsm library outside checkTransition function.